pkgs/top-level/stage.nix

```diff
@@ -206,7 +206,11 @@ let
         name: _: throw "${name} ${type} is not available, please enable config.${configOption}"
       ) pkgs;
 
-  aliases = self: super: lib.optionalAttrs config.allowAliases (import ./aliases.nix lib self super);
+  aliases = mkOptionalOverlay {
+    configOption = "allowAliases";
+    type = "package";
+    overlay = import ./aliases.nix lib;
+  };
```

Comment:
Given how simple this was before I'm not sure the quite complex mkOptionalOverlay function is worth it?

Reply:
Ok, I've dropped the commit which did that. We can figure something out later.